2-completed-app/AzureTablesDemoApplication/src/main/java/com/data/tables/service/impl/TablesServiceImpl.java

```diff
@@ -0,0 +1,144 @@
+import com.data.tables.models.SampleDataInputModel;
+import com.data.tables.models.WeatherDataModel;
+import com.data.tables.models.WeatherInputModel;
+import com.data.tables.service.ITablesService;
+import com.data.tables.untils.WeatherDataUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.util.StringUtils;
+
+
+@Service
+public class TablesServiceImpl implements ITablesService {
+
+    @Autowired
+    private TableClient tableClient;
+    /**
+     * Get all weather data entities
+     * @return List<WeatherDataModel>
+     */
+    @Override
+    public List<WeatherDataModel> getAllRows() {
+        List<WeatherDataModel> modelList = new ArrayList<>();
+        tableClient.listEntities().stream().forEach(entity -> modelList.add(WeatherDataUtils.mapTableEntityToWeatherDataModel(entity)));
+        return Collections.unmodifiableList(WeatherDataUtils.filledValue(modelList));
+    }
+
+    /**
+     * Get weather data entities by filters
+     * @return List<WeatherDataModel>
+     */
+    @Override
+    public List<WeatherDataModel> getFilteredRows(FilterResultsInputModel model) {
+
+        List<String> filters = new ArrayList<>();
+
+        if (!StringUtils.isEmptyOrWhitespace(model.getPartitionKey())) {
+            filters.add(String.format("PartitionKey eq '%s'", model.getPartitionKey()));
+        }
+        if (!StringUtils.isEmptyOrWhitespace(model.getRowKeyDateStart())
+                && !StringUtils.isEmptyOrWhitespace(model.getRowKeyTimeStart())) {
+            filters.add(String.format("RowKey ge '%s %s'", model.getRowKeyDateStart(), model.getRowKeyTimeStart()));
+        }
+        if (!StringUtils.isEmptyOrWhitespace(model.getRowKeyDateEnd())
+                && !StringUtils.isEmptyOrWhitespace(model.getRowKeyTimeEnd())) {
+            filters.add(String.format("RowKey le '%s %s'", model.getRowKeyDateEnd(), model.getRowKeyTimeEnd()));
+        }
+        if (model.getMinTemperature() != null) {
+            filters.add(String.format("Temperature ge %f", model.getMinTemperature()));
+        }
+        if (model.getMaxTemperature() != null) {
+            filters.add(String.format("Temperature le %f", model.getMaxTemperature()));
+        }
+        if (model.getMinPrecipitation() != null) {
+            filters.add(String.format("Precipitation ge %f", model.getMinPrecipitation()));
+        }
+        if (model.getMaxPrecipitation() != null) {
+            filters.add(String.format("Precipitation le %f", model.getMaxPrecipitation()));
+        }
+
+        List<WeatherDataModel> modelList = new ArrayList<>();
+        tableClient.listEntities(new ListEntitiesOptions().setFilter(String.join(" and ", filters)), null, null)
+                .stream().forEach(entity -> modelList.add(WeatherDataUtils.mapTableEntityToWeatherDataModel(entity)));
+        return Collections.unmodifiableList(WeatherDataUtils.filledValue(modelList));
```

Comment:
nit; I also think we can make this look a bit cleaner:
```suggestion
        List<WeatherDataModel> modelList = tableClient.listEntities(new ListEntitiesOptions()
                .setFilter(String.join(" and ", filters)), null, null).stream()
            .map(WeatherDataUtils::mapTableEntityToWeatherDataModel)
            .collect(Collectors.toList());
        return Collections.unmodifiableList(WeatherDataUtils.filledValue(modelList));
```